lib/query/ViewQuery/Dimensions.js

```diff
@@ -10,8 +11,18 @@ class Dimensions {
     const resultDimensions = view.out(ns.view.dimension)
     const filterDimensions = view.out(ns.view.filter).out(ns.view.dimension)
 
+    this.resolveProperty = function (dimensionPtr) {
+      if (dimensionPtr.out(ns.view.as).term) {
+        return dimensionPtr.out(ns.view.as).term
+      } else if (dimensionPtr.out(ns.view.from).out(ns.view.path).term) {
+        return dimensionPtr.out(ns.view.from).out(ns.view.path).term
+      } else {
+        return rdf.blankNode()
+      }
+    }
```

Comment:
Interesting choice to return blank node. No issue with that if it does not cause any problems. 👌 

About this "method" itself, since it does not even use the `this`, I would simply put it outside the class

```js
class Dimensions {
  // ...
}

function resolveProperty(dimensionPtr) {
  return ...
}
```